src/clients/c++/perf_client.cc

```diff
@@ -135,14 +135,30 @@ using TimestampVector =
 namespace {
 
 volatile bool early_exit = false;
+// The flag is set when there are no in-flight inferences to complete.
+volatile bool finished_inferencing = false;
 
 void
 SignalHandler(int signum)
 {
-  std::cout << "Interrupt signal (" << signum << ") received." << std::endl
-            << "Waiting for in-flight inferences to complete." << std::endl;
-
-  early_exit = true;
+  // Upon invoking the SignalHandler for the first time early_exit flag is
+  // invoked. On the subsequent invocations, if there are no in-flight inference
+  // requests, the program exits.
+  if (!early_exit) {
+    std::cout << "Interrupt signal (" << signum << ") received." << std::endl
```

Comment:
Can you just unconditionally print "Interrupt signal ..."  at the start of SignalHandler